src/components/CheckBoxFilterContainer.ts

```diff
@@ -0,0 +1,203 @@
+    unCheckedAttributeValue: string;
+    unCheckedConstraint: string;
+}
+
+type FilterOptions = "attribute" | "XPath" | "None";
+interface HybridConstraint {
+    attribute: string;
+    operator: string;
+    value: string;
+    path?: string;
+}
+
+export interface ListView extends mxui.widget._WidgetBase {
+    _datasource: {
+        _constraints: HybridConstraint[] | string;
+        _pageObjs: mendix.lib.MxObject[];
+    };
+    datasource: {
+        type: "microflow" | "entityPath" | "database" | "xpath";
+    };
+    __customWidgetDataSourceHelper: DataSourceHelper;
+    update: (obj: mendix.lib.MxObject | null, callback?: () => void) => void;
+    _entity: string;
+}
+
+export interface ContainerState {
+    listViewAvailable: boolean;
+    targetListView?: ListView;
+    targetNode?: HTMLElement;
+    validationPassed?: boolean;
+    alertMessage?: string;
+}
+
+export default class CheckboxFilterContainer extends Component<ContainerProps, ContainerState> {
+    private dataSourceHelper: DataSourceHelper;
+    private alertMessage: string;
+    constructor(props: ContainerProps) {
+        super(props);
+
+        this.applyFilter = this.applyFilter.bind(this);
+        if (!Utils.validateProps(props)) {
+            // Ensures that the listView is connected so the widget doesn't break in mobile due to unpredictable render timing
+            dojoConnect.connect(props.mxform, "onNavigation", this, this.connectToListView.bind(this));
+        }
+
+        this.state = { listViewAvailable: false };
+    }
+
+    render() {
+        return createElement("div",
+            {
+                className: classNames("widget-checkbox-filter", this.props.class),
+                style: parseStyle(this.props.style)
+            },
+            this.renderAlert(),
+            this.renderComponent()
+        );
+    }
+
+    componentWillUpdate(nextProps: ContainerProps, nextState: ContainerState) {
+        console.log("willUpdate"); // tslint:disable-line
+        this.alertMessage = this.validate(nextProps, nextState.targetListView);
+        this.applyFilter(nextProps.defaultChecked);
+    }
+
+    componentDidUpdate(_prevProps: ContainerProps, prevState: ContainerState) {
+        if (this.state.listViewAvailable && !prevState.listViewAvailable) {
+            this.applyFilter(this.props.defaultChecked);
+        }
+    }
+
+    private renderAlert() {
+        return createElement(Alert, {
+            className: "widget-checkbox-filter-alert",
+            message: this.alertMessage
+        });
+    }
+
+    private renderComponent(): ReactElement<CheckboxFilterProps> {
+        if (!this.alertMessage) {
+            return createElement(CheckboxFilter, {
+                handleChange: this.applyFilter,
+                isChecked: this.props.defaultChecked
+            });
+        }
+
+        return null;
+    }
+
+    private applyFilter(isChecked: boolean) {
+        // construct constraint based on checked
```

Comment:
Comments sentences should always start with a capital

Reply:
Done